### PR TITLE
Refactoring for star schema

### PR DESCRIPTION
**Pull request by GabeLoins**

### CODE REVIEW

**Comment by GabeLoins** [comment]
@vqtran 

I'm not sure exactly how were going to merge this, I might need your help but this is the general idea.

**Comment by vqtran** [comment]
Actually what's wrong with just taking all of these changes over what we previously had?

**Comment by GabeLoins** [comment]
like incorporating it with your count module

**Comment by vqtran** [comment]
You mean the thing I'm doing with abstracting out how we represent queries / make one intent handle many aggregates? That's not near a PR yet I think, I would take all of these changes first. Also it would make sense for what I was planning to be built on top of querier.

File: src/main/java/echoquery/intents/CountHandler.java [hunk 1]
```diff
@@ -1,40 +1,35 @@
...
-  private final Connection conn;
-  private final Logger log;
+  private static final Logger log =
+      LoggerFactory.getLogger(CountHandler.class);
+  private static final StarSchemaQuerier querier = StarSchemaQuerier.getInstance();
 
-  public CountHandler(Connection conn, Logger log) {
-    this.conn = conn;
-    this.log = log;
+  public CountHandler() {
```
Comment (vqtran) [inline src/main/java/echoquery/intents/CountHandler.java]: Constructor not needed anymore right?

File: src/main/java/echoquery/sql/ColumnValue.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package echoquery.sql;
+
+public class ColumnValue {
+  private String value;
+  private Boolean isString;
```
Comment (vqtran) [inline src/main/java/echoquery/sql/ColumnValue.java]: Boolean -> boolean

File: src/main/java/echoquery/sql/ColumnValue.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+  public ColumnValue(String value, Boolean isString) {
```
Comment (vqtran) [inline src/main/java/echoquery/sql/ColumnValue.java]: Boolean -> boolean

File: src/main/java/echoquery/sql/InvalidJoinRecipe.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package echoquery.sql;
+
+public class InvalidJoinRecipe implements JoinRecipe {
+
+  public InvalidJoinRecipe() {
```
Comment (vqtran) [inline src/main/java/echoquery/sql/InvalidJoinRecipe.java]: Unnecessary constructor

File: src/main/java/echoquery/sql/JoinRecipe.java [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package echoquery.sql;
+
+public interface JoinRecipe {
```
Comment (vqtran) [inline src/main/java/echoquery/sql/JoinRecipe.java]: Comment for what a join recipe is / what render does

File: src/main/java/echoquery/sql/StarSchemaStore.java [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * 
+ * @author gabe
+ * Knows the different columns on each table and the connections between tables
+ *
+ */
+public class StarSchemaStore {
```
Comment (vqtran) [inline src/main/java/echoquery/sql/StarSchemaStore.java]: Yeah SchemaStore here would be the better name as you said

File: src/main/java/echoquery/sql/StarSchemaQuerier.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+import java.sql.Statement;
+import java.sql.DriverManager;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import echoquery.utils.EchoQueryCredentials;
+
+
+public class StarSchemaQuerier {
```
Comment (vqtran) [inline src/main/java/echoquery/sql/StarSchemaQuerier.java]: Is this actually star schema specific either? SchemaQuerier?

File: src/main/java/echoquery/sql/StarSchemaQuerier.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+  private static final StarSchemaQuerier querier = new StarSchemaQuerier();
+  private static final Logger log =
+      LoggerFactory.getLogger(StarSchemaQuerier.class);
+  private final Connection conn = this.getConnection();
```
Comment (vqtran) [inline src/main/java/echoquery/sql/StarSchemaQuerier.java]: Confused how this actually even works, "this" is referenced before the constructor is even called?

File: src/main/java/echoquery/sql/StarSchemaQuerier.java [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+  private final StarSchemaStore tableStore;
+
+  private Connection getConnection() {
```
Comment (vqtran) [inline src/main/java/echoquery/sql/StarSchemaQuerier.java]: Lets keep this at the top level file, and pass it down into the constructor

File: src/main/java/echoquery/sql/StarSchemaStore.java [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+        // fill in table to foreign keys
+        adjustTableToForeignKeys(table, md);
+      }
+    } catch (SQLException e) {
+      log.info("Error generating the StarSchemaStore");
+      e.printStackTrace();
+    }
+  }
+  
+  private void adjustTableToForeignKeys(String table, DatabaseMetaData md) throws SQLException {
```
Comment (vqtran) [inline src/main/java/echoquery/sql/StarSchemaStore.java]: Line length things

**Comment by vqtran** [comment]
Let me know if you don't have a chance to make these changes - I might be able to pull this branch and work on this tonight because APMA turned out to be kind of chill.

File: pom.xml [hunk 11]
```diff
@@ -88,6 +88,11 @@
       <artifactId>aws-java-sdk-dynamodb</artifactId>
       <version>1.9.40</version>
     </dependency>
+    <dependency>
```
Comment (vqtran) [inline pom.xml]: oops ignore this

**Pull request merged**